vimeo/VimeoUpload
# CI-112 Unit testing

#### Ticket
**Required for Vimeans only**
[CI-112](https://vimean.atlassian.net/browse/CI-112)

#### Ticket Summary
Implementation of tooling to run unit tests an report results.

#### Implementation Summary
- Added test targets to Podfile, and committed corresponding Pods changes (from `bundle exec pod install`)
- Added `if-check` in VimeoUpload-iOS-OldUpload/AppDelegate.swift to only ask for camera roll permissions if not running as a test. This is to avoid complications with systems notifications when running tests in the simulator.
- Added Danger to handle test result reporting

#### How to Test
- Fastlane should be able to run tests for both schemes - VimeoUpload-iOS & VimeoUpload-iOS-OldUpload via `bundle exec fastlane test`. Tests should launch and run in the default simulator both locally and on buildbot.
- If the build is part of a PR, the test results should be reported to the PR comments via Danger.
